include/elements/2d/quadrilateral_bspline_element.tcc

```diff
@@ -0,0 +1,339 @@
+//! Assign nodal connectivity property for bspline elements
+template <unsigned Tdim, unsigned Tpolynomial>
+void mpm::QuadrilateralBSplineElement<Tdim, Tpolynomial>::
+    initialise_bspline_connectivity_properties(
+        const Eigen::MatrixXd& nodal_coordinates,
+        const std::vector<std::vector<unsigned>>& nodal_properties) {
+  assert(nodal_coordinates.rows() == nodal_properties.size());
+
+  this->nconnectivity_ = nodal_coordinates.rows();
+  this->nodal_coordinates_ = nodal_coordinates;
+  this->node_type_ = nodal_properties;
+
+  //! Uniform spacing length in 2D
+  this->spacing_length_ =
+      std::abs(nodal_coordinates(1, 0) - nodal_coordinates(0, 0));
+}
+
+//! Return shape functions of a Quadrilateral BSpline Element at a given
+//! local coordinate
```

Comment:
How about unifying the expression to "local coordinate" or "natural coordinate" in the whole code?

Reply:
Hmmm... maybe we do that in a separate cleanup PR? Just for clarity right?